client/task-list/tasks/payments/RemotePayments/components/PaymentConnect.js

```diff
@@ -103,39 +137,45 @@ export const PaymentConnect = ( {
 		},
 	} );
 
-	return (
-		<Form
-			initialValues={ getInitialConfigValues() }
+	const DefaultSettings = ( { ...props } ) => (
+		<Settings
+			fields={ fields }
+			isBusy={ isOptionsRequesting }
 			onSubmitCallback={ updateSettings }
+			onButtonClickCallback={ () => recordConnectStartEvent( key ) }
+			buttonLabel={ __( 'Proceed', 'woocommerce-admin' ) }
 			validate={ validate }
-		>
-			{ ( { getInputProps, handleSubmit } ) => {
-				return (
-					<>
-						{ ( fields || [] ).map( ( field ) => (
-							<TextControl
-								key={ field.name }
-								label={ field.label }
-								required
-								{ ...getInputProps( field.name ) }
-							/>
-						) ) }
-
-						<Button
-							isPrimary
-							isBusy={ isOptionsRequesting }
-							onClick={ ( event ) => {
-								recordConnectStartEvent( key );
-								handleSubmit( event );
-							} }
-						>
-							{ __( 'Proceed', 'woocommerce-admin' ) }
-						</Button>
-
-						<p>{ helpText }</p>
-					</>
-				);
-			} }
-		</Form>
+			{ ...props }
+		/>
+	);
+
+	if ( state === 'error' ) {
+		return (
+			<Text>
+				{ __( 'There was an error loading the payment fields' ) }
+			</Text>
+		);
+	}
+
+	if ( state === 'loading' ) {
+		return <Spinner />;
+	}
+
+	return (
+		<>
+			{ hasFills ? (
+				<WooRemotePaymentSettings.Slot
+					defaultSettings={ DefaultSettings }
+					defaultUpdate={ updateSettings }
+					markConfigured={ () => markConfigured( key ) }
+					id={ key }
```

Comment:
You could also pass in `fields` here as well under `defaultFields` so the extension can decide if they want to use `DefaultSettings` or create whatever else they want.